src/coprocessor/dag/rpn_expr/impl_compare.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Ordering;
+
+use super::types::RpnFnCallPayload;
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::{codec, Error, Result};
+
+pub struct RpnFnCompare<C: Comparer> {
+    _phantom: std::marker::PhantomData<C>,
+}
+
+impl<C: Comparer> RpnFnCompare<C> {
+    #[inline]
+    pub fn new() -> Self {
+        Self {
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        lhs: &Option<C::T>,
+        rhs: &Option<C::T>,
+    ) -> Result<Option<i64>> {
+        C::compare(lhs, rhs)
+    }
+}
+
+impl<C: Comparer> std::fmt::Debug for RpnFnCompare<C> {
+    #[inline]
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "RpnFnCompare")
+    }
+}
+
+// See rust-lang/rust#26925 for why the followings are implemented manually. =====
+
+impl<C: Comparer> Copy for RpnFnCompare<C> {}
+
+impl<C: Comparer> Clone for RpnFnCompare<C> {
+    #[inline]
+    fn clone(&self) -> Self {
+        Self::new()
+    }
+}
+
+unsafe impl<C: Comparer> Send for RpnFnCompare<C> {}
+
+unsafe impl<C: Comparer> Sync for RpnFnCompare<C> {}
```

Comment:
Why would there be any `C` which is not `Send + Sync`?

Reply:
Good idea

Reply:
For example, Rust cannot infer `Send + Sync` for `BasicComparer<T: Evaluable + Ord, F: CmpOp>` because `Evaluable` is not `Send + Sync`.

Reply:
Okay, but why Evaluable doesn't require Sync in the first place? (It already requires Clone + Debug + Send + 'static)